scripts/build.py

```diff
@@ -38,8 +36,13 @@ def detect_arch():
                 arch["libdir"] = os.path.expanduser(arch["libdir"])
             else:
                 global tmpdir
-                tmpdir = tempfile.TemporaryDirectory()
-                arch["libdir"] = tmpdir.name
+                if "CI" in os.environ:
+                    tmpdir = "./tmplibdir"
+                    os.makedirs(tmpdir, exist_ok=True)
+                    arch["libdir"] = tmpdir
+                else:
+                    tmpdir = tempfile.TemporaryDirectory()
+                    arch["libdir"] = tmpdir.name
```

Comment:
The `/tmp` dir was being mysteriously cleaned during building, so it was changed to a local directory instead. This should be revisited later.